lib/server.js

```diff
@@ -107,7 +107,7 @@ function Server (torrent, opts) {
         return serveIndexPage()
       }
 
-      var index = Number(pathname.slice(1))
+      var index = Number(pathname.split('/')[0])
```

Comment:
I am running this on my own deployment and it is working. Not sure how to explain this though, your logic is also correct. I'll do some more testing and update later.